code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -14,49 +15,700 @@
+				rglass_module.glasource = get_or_create_estorage(rglass_module.glasource)
+
+		if(istype(sheet_module.source))
+			sheet_module.cost = max(sheet_module.cost, 1) // Must not cost 0 to prevent div/0 errors.
+			sheet_module.is_cyborg = TRUE
+
+	if(I.loc != src)
+		I.forceMove(src)
+	modules += I
+	ADD_TRAIT(I, TRAIT_NODROP, CYBORG_ITEM_TRAIT)
+	I.mouse_opacity = MOUSE_OPACITY_OPAQUE
+	if(nonstandard)
+		added_modules += I
+	if(requires_rebuild)
+		rebuild_modules()
+	return I
+
+/obj/item/robot_model/proc/remove_module(obj/item/I, delete_after)
+	basic_modules -= I
+	modules -= I
+	emag_modules -= I
+	added_modules -= I
+	rebuild_modules()
+	if(delete_after)
+		qdel(I)
+
+/obj/item/robot_model/proc/rebuild_modules() //builds the usable module list from the modules we have
+	var/mob/living/silicon/robot/R = loc
+	var/list/held_modules = R.held_items.Copy()
+	var/active_module = R.module_active
+	R.uneq_all()
+	modules = list()
+	for(var/obj/item/I in basic_modules)
+		add_module(I, FALSE, FALSE)
+	if(R.emagged)
+		for(var/obj/item/I in emag_modules)
+			add_module(I, FALSE, FALSE)
+	for(var/obj/item/I in added_modules)
+		add_module(I, FALSE, FALSE)
+	for(var/i in held_modules)
+		if(i)
+			R.equip_module_to_slot(i, held_modules.Find(i))
+	if(active_module)
+		R.select_module(held_modules.Find(active_module))
+	if(R.hud_used)
+		R.hud_used.update_robot_modules_display()
+
+/obj/item/robot_model/proc/respawn_consumable(mob/living/silicon/robot/R, coeff = 1)
+	for(var/datum/robot_energy_storage/st in storages)
+		st.energy = min(st.max_energy, st.energy + coeff * st.recharge_rate)
+
+	for(var/obj/item/I in get_usable_modules())
+		if(istype(I, /obj/item/assembly/flash))
+			var/obj/item/assembly/flash/F = I
+			F.times_used = 0
+			F.burnt_out = FALSE
+			F.update_appearance()
+		else if(istype(I, /obj/item/melee/baton/security))
+			var/obj/item/melee/baton/security/baton = I
+			baton.cell?.charge = baton.cell.maxcharge
+		else if(istype(I, /obj/item/gun/energy))
+			var/obj/item/gun/energy/EG = I
+			if(!EG.chambered)
+				EG.recharge_newshot() //try to reload a new shot.
+
+	R.toner = R.tonermax
+
+/obj/item/robot_model/proc/get_or_create_estorage(storage_type)
+	return (locate(storage_type) in storages) || new storage_type(src)
+
+/obj/item/robot_model/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_CONTENTS)
+		return
+	for(var/obj/O in modules)
+		O.emp_act(severity)
+	..()
+
+/obj/item/robot_model/proc/transform_to(new_config_type, forced = FALSE)
+	var/mob/living/silicon/robot/R = loc
+	var/obj/item/robot_model/RM = new new_config_type(R)
+	RM.robot = R
+	if(!RM.be_transformed_to(src, forced))
+		qdel(RM)
+		return
+	R.model = RM
+	R.update_module_innate()
+	RM.rebuild_modules()
+	R.radio.recalculateChannels()
+
+	INVOKE_ASYNC(RM, .proc/do_transform_animation)
+	qdel(src)
+	return RM
+
+/obj/item/robot_model/proc/be_transformed_to(obj/item/robot_model/old_model, forced = FALSE)
```

Comment:
space out this proc some so its not a solid block

Reply:
we dont do that in the fikou household